src/services/pool.ts

```diff
@@ -89,13 +89,25 @@ export const findAllRepos = async (options: FindOptions = {}, reqId?: string) =>
 
 const getRepoUrl = (baseUrl: string, prefix: string) => assertTrailingSlash(baseUrl) + assertTrailingSlash(prefix);
 
-const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) =>
-  branches.map((branchEntry) => ({
+/**
+ *
+ * branch full url util,
+ * It dynamically handling url slugs for
+ * repos that have more then one version
+ */
+const setBranchFullUrl = (branchEntry: BranchEntry, fullBaseUrl: string, useUrlSlug = true) => {
+  const urlSlug = useUrlSlug && branchEntry?.usrSlug ? branchEntry.urlSlug : '';
```

Comment:
out of curiosity, is this in line with determination logic for CDN pertaining to repos_branches?

Reply:
@seungpark can you elaborate a bit more on this, sorry I don't think I am fully tracking.